api/client/lib/client.go

```diff
@@ -30,13 +31,19 @@ type Client struct {
 	version string
 	// custom http headers configured by users
 	customHTTPHeaders map[string]string
+	// a cookie jar, if we're meant to be using cookies
+	jar *cookiejar.Jar
+	// authResponders provide ways to authenticate to servers
+	authResponders *map[string]*[]authResponder
```

Comment:
We should probably maintain consistency with the current parameters and do a `map[string]*[]authReponder`.